irohad/ametsuchi/block_query.hpp

```diff
@@ -23,27 +23,28 @@
 #include <rxcpp/rx-observable.hpp>
 
 #include "common/types.hpp"
+#include "interfaces/iroha_internal/block.hpp"
+#include "interfaces/transaction.hpp"
 
 namespace iroha {
 
-  namespace model {
-    struct Transaction;
-    struct Block;
-  }
-
   namespace ametsuchi {
     /**
      * Public interface for queries on blocks and transactions
      */
     class BlockQuery {
      public:
+      using wTransaction =
+          std::shared_ptr<shared_model::interface::Transaction>;
+      using wBlock = std::shared_ptr<shared_model::interface::Block>;
+
       virtual ~BlockQuery() = default;
       /**
        * Get all transactions of an account.
        * @param account_id - account_id (accountName@domainName)
        * @return observable of Model Transaction
        */
-      virtual rxcpp::observable<model::Transaction> getAccountTransactions(
+      virtual rxcpp::observable<wTransaction> getAccountTransactions(
           const std::string &account_id) = 0;
```

Comment:
std::string -> `shared_model::interface::types::AccountIdType`

Same can be done to other argument types in this file.